docs/docsite/rst/user_guide/playbooks_filters.rst

```diff
@@ -536,61 +517,235 @@ Now, let's take the following data structure::
         }
     }
 
-To extract all clusters from this structure, you can use the following query::
+To extract all clusters from this structure, you can use the following query::
+
+    - name: "Display all cluster names"
+      debug:
+        var: item
+      loop: "{{ domain_definition | json_query('domain.cluster[*].name') }}"
+
+Same thing for all server names::
+
+    - name: "Display all server names"
+      debug:
+        var: item
+      loop: "{{ domain_definition | json_query('domain.server[*].name') }}"
+
+This example shows ports from cluster1::
+
+    - name: "Display all ports from cluster1"
+      debug:
+        var: item
+      loop: "{{ domain_definition | json_query(server_name_cluster1_query) }}"
+      vars:
+        server_name_cluster1_query: "domain.server[?cluster=='cluster1'].port"
+
+.. note:: You can use a variable to make the query more readable.
+
+Or, alternatively print out the ports in a comma separated string::
+
+    - name: "Display all ports from cluster1 as a string"
+      debug:
+        msg: "{{ domain_definition | json_query('domain.server[?cluster==`cluster1`].port') | join(', ') }}"
+
+.. note:: Here, quoting literals using backticks avoids escaping quotes and maintains readability.
+
+Or, using YAML `single quote escaping <https://yaml.org/spec/current.html#id2534365>`_::
+
+    - name: "Display all ports from cluster1"
+      debug:
+        var: item
+      loop: "{{ domain_definition | json_query('domain.server[?cluster==''cluster1''].port') }}"
+
+.. note:: Escaping single quotes within single quotes in YAML is done by doubling the single quote.
+
+In this example, we get a hash map with all ports and names of a cluster::
+
+    - name: "Display all server ports and names from cluster1"
+      debug:
+        var: item
+      loop: "{{ domain_definition | json_query(server_name_cluster1_query) }}"
+      vars:
+        server_name_cluster1_query: "domain.server[?cluster=='cluster2'].{name: name, port: port}"
+
+
+Randomizing data
+================
+
+When you need a randomly generated value, use one of these filters.
+
+.. _random_mac_filter:
+
+Random MAC addresses
+--------------------
+
+.. versionadded:: 2.6
+
+This filter can be used to generate a random MAC address from a string prefix.
+
+To get a random MAC address from a string prefix starting with '52:54:00'::
+
+    "{{ '52:54:00' | random_mac }}"
+    # => '52:54:00:ef:1c:03'
+
+Note that if anything is wrong with the prefix string, the filter will issue an error.
+
+As of Ansible version 2.9, it's also possible to initialize the random number generator from a seed. This way, you can create random-but-idempotent MAC addresses::
+
+    "{{ '52:54:00' | random_mac(seed=inventory_hostname) }}"
+
+.. _random_filter:
+
+Random items or numbers
+-----------------------
+
+.. versionadded:: 1.6
+
+This filter can be used similar to the default jinja2 random filter (returning a random item from a sequence of
```

Comment:
```suggestion
This filter can be used similar to the default Jinja2 random filter (returning a random item from a sequence of
```

Reply:
obviously, the further along I get, the more nitpicky... another TIL - Jinja2 is initial Cap according to their documentation...

Reply:
Thanks for finding the nits!